db/indexes.go

```diff
@@ -330,30 +300,29 @@ func (i *SGIndex) createIfNeeded(ctx context.Context, bucket base.N1QLStore, opt
 
 	// start a retry loop to create index,
 	worker := func() (shouldRetry bool, err error, value interface{}) {
-		err = bucket.CreateIndex(ctx, indexName, indexExpression, filterExpression, n1qlOptions)
+		err = bucket.CreateIndexIfNotExists(ctx, indexName, indexExpression, filterExpression, n1qlOptions)
 		if err != nil {
-			// If index has already been created (race w/ other SG node), return without error
-			if err == base.ErrAlreadyExists {
-				isDeferred = false // Index already exists, don't need to update.
-				return false, nil, nil
-			}
 			if strings.Contains(err.Error(), "not enough indexer nodes") {
 				return false, fmt.Errorf("Unable to create indexes with the specified number of replicas (%d).  Increase the number of index nodes, or modify 'num_index_replicas' in your Sync Gateway database config.", options.NumReplicas), nil
+			} else if errors.Is(err, base.ErrIndexBackgroundRetry) {
+				// bubble up without retry so the caller can handle waiting for index readiness
+				base.DebugfCtx(ctx, base.KeyQuery, "Index creation failed with background retry error - will wait for index readiness: %v", err)
```

Comment:
We should probably log the name of the index here, and maybe change the comment, to indicate that this was a failure returned by N1QL but that the index is expected to be created on the server.